hil/cli.py

```diff
@@ -151,94 +97,54 @@ def setup_http_client():
     C = Client(ep, http_client)
 
 
-def check_status_code(response):
-    """Check the status code of the response.
-
-    If it is a successful status code, print the body of the response to
-    stdout. Otherwise, print an error message, and raise a
-    FailedAPICallException.
-    """
-    if response.status_code < 200 or response.status_code >= 300:
-        sys.stderr.write('Unexpected status code: %d\n' % response.status_code)
-        sys.stderr.write('Response text:\n')
-        sys.stderr.write(response.content + "\n")
-        raise FailedAPICallException()
-    else:
-        sys.stdout.write(response.content + "\n")
-
-# Function object_url should be DELETED.
-
-
-def object_url(*args):
-    """Return a url with a prefix of the HIL endpoint, and args as the
-    (remaining) segments of the path.
-
-    TODO: This function's name is no longer very accurate.  As soon as it is
-    safe, we should change it to something more generic.
-    """
-    # Prefer an environmental variable for getting the endpoint if available.
-    url = os.environ.get('HIL_ENDPOINT')
-    if url is None:
-        config.setup()
-        url = cfg.get('client', 'endpoint')
+###############################
+### First level subcommands ###
+###############################
 
-    for arg in args:
-        url += '/' + urllib.quote(arg, '')
-    return url
+@click.group()
+def cli():
+    """The HIL Command line"""
 
+@cli.group()
+def node():
+    """Commands related to node"""
 
-# Helper functions for making HTTP requests against the API.
-#    Uses the global variable `http_client` to make the request.
-#
-#    Arguments:
-#
-#        `url` - The url to make the request to
-#        `data` - the body of the request (for PUT, POST and DELETE)
-#        `params` - query parameters (for GET)
+@cli.group()
+def project():
+    """Commands related to project"""
 
-def do_put(url, data={}):
-    """do a put request and check the response."""
-    check_status_code(http_client.request('PUT', url, data=json.dumps(data)))
+@cli.group()
```

Comment:
it is kinda hard to see so please forgive me if I said something crazy. 

Are we having two `project` under `cli.group()` here?

Reply:
no, there's just one? I defined all the top level subcommands on top.

Reply:
lol, is this confused face new? I like it ahha